Modules/_io/fileio.c

```diff
@@ -455,11 +458,17 @@ _io_FileIO___init___impl(fileio *self, PyObject *nameobj, const char *mode,
 #endif
     }
 
-    PyMem_Free(self->stat_atopen);
```

Comment:
Not sure which is preferred here. The Free + New back to back feels weird, but version with the conditional feels like "extra work/code that _should_ never be executed" (but that the compiler / tools that run currently won't validate...).

Reply:
Maybe `if (ptr != NULL) free(ptr)` would be more regular and avoid the 4 lines long comment.

Reply:
Moved to that, dropped comment and simplified / back to always calling PyMem_New